gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -738,48 +538,62 @@ def _convert_to_variant(self, record):
   def _get_variant_info(self, record):
     # type: (libcbcf.VariantRecord) -> Dict[str, Any]
     info = {}
-    for k, v in list(record.info.items()):
-      if k != END_INFO_KEY:
-        if isinstance(v, tuple):
-          info[k] = list(map(self._convert_field, v))
+    for info_id, field in list(record.info.items()):
+      # End position is extracted from record.
+      if info_id == END_INFO_KEY:
+        continue
+
+      # Check if field is a flag:
+      # - If it's defined in the header, it always has value True.
+      # - If it's not defined in the header, it would be None.
+      #
+      # Note that if type was mismatched (eg. string provided for an int field),
+      # the field would also be None, so header presence needs to be verified.
+      if isinstance(field, bool) or (field is None and
+                                     info_id not in self._original_info_list):
+        info[info_id] = True
+      else:
+        if isinstance(field, tuple):
+          info[info_id] = self._homogenize_list_dtype(
+              map(self._convert_field, field, [True] * (len(field))))
         else:
-          # If a field was not provided in the header, make it a list to
-          # follow the PyVCF standards.
-          info[k] = (
-              self._convert_field(v) if k in self._original_info_list else
-              [self._convert_field(v)])
+          # If a field was not provided in the header, make it a list.
+          info[info_id] = (self._convert_field(field, True) if
+                           info_id in self._original_info_list else
+                           [self._convert_field(field, True)])
```

Comment:
Thanks for updating this method, code is more clear now.

Now my question is why we need to attempt to convert singular values (non tuples) to integer?